From 8d25fd1aaed7f7fbca44e7f985cdd430af0209c8 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Wed, 8 Aug 2018 19:41:07 +0300
Subject: [PATCH 0548/1051] net: mvpp2: xps mapping for tx queues

Map tx-queues to hifs (e.g. to CPUs).
Number of mapped queues nay be less than "ntxqs".

Change-Id: I07e9d57ac1dee2197397ce8059ed40062a882cc8
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/59477
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: Stefan Chulski <stefanc@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 18432b1f3f89..efbd1273fa8e 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -2425,6 +2425,10 @@ static int mvpp2_setup_txqs(struct mvpp2_port *port)
 		netif_set_xps_queue(port->dev, cpumask_of(cpu), queue);
 	}
 
+	/* XPS mapping queues to 0..N cpus (may be less than ntxqs) */
+	for_each_online_cpu(cpu)
+		netif_set_xps_queue(port->dev, cpumask_of(cpu), cpu);
+
 	if (port->has_tx_irqs) {
 		mvpp2_tx_time_coal_set(port);
 		for (queue = 0; queue < port->ntxqs; queue++) {
-- 
2.17.1

